core/src/main/java/feast/core/model/EntityV2.java

```diff
@@ -0,0 +1,136 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.model;
+
+import com.google.protobuf.Timestamp;
+import feast.core.util.TypeConversion;
+import feast.proto.core.EntityProto;
+import feast.proto.core.EntityProto.*;
+import feast.proto.types.ValueProto.ValueType;
+import java.util.Map;
+import javax.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@javax.persistence.Entity
+@Table(
+    name = "entities_v2",
+    uniqueConstraints = @UniqueConstraint(columnNames = {"name", "project_name"}))
+public class EntityV2 extends AbstractTimestampEntity {
+  @Id @GeneratedValue private long id;
+
+  // Name of the Entity
+  @Column(name = "name", nullable = false)
+  private String name;
+
+  // Project that this Entity belongs to
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "project_name")
+  private Project project;
+
+  // Description of entity
+  @Column(name = "description", columnDefinition = "text")
+  private String description;
+
+  // Columns of entities
+  /** Data type of each entity column: String representation of {@link ValueType} * */
+  private String type;
+
+  // User defined metadata
+  @Column(name = "labels", columnDefinition = "text")
+  private String labels;
+
+  public EntityV2() {
+    super();
+  }
+
+  public EntityV2(
+      String name, String description, ValueType.Enum type, Map<String, String> labels) {
+    this.name = name;
+    this.description = description;
+    this.type = type.toString();
+    this.labels = TypeConversion.convertMapToJsonString(labels);
+  }
+
+  public static EntityV2 fromProto(EntityProto.Entity entityProto) {
+    EntitySpecV2 spec = entityProto.getSpec();
+
+    return new EntityV2(
+        spec.getName(), spec.getDescription(), spec.getValueType(), spec.getLabelsMap());
+  }
+
+  public EntityProto.Entity toProto() {
+    EntityMeta.Builder meta =
+        EntityMeta.newBuilder()
+            .setCreatedTimestamp(
+                Timestamp.newBuilder().setSeconds(super.getCreated().getTime() / 1000L));
+
+    EntitySpecV2.Builder spec =
+        EntitySpecV2.newBuilder()
+            .setName(getName())
+            .setDescription(getDescription())
+            .setValueType(ValueType.Enum.valueOf(getType()))
+            .putAllLabels(TypeConversion.convertJsonStringToMap(labels));
+
+    // Build Entity
+    EntityProto.Entity entity = EntityProto.Entity.newBuilder().setMeta(meta).setSpec(spec).build();
+    return entity;
+  }
+
+  /**
+   * Updates the existing entity from a proto.
```

Comment:
Does this only update labels?

Reply:
It should update description as well. Will add.

Reply:
Updated.